isis/src/lro/apps/lronacpho/main.cpp

```diff
@@ -169,24 +166,74 @@ void phoCal(Buffer &in, Buffer &out){
  * @param in Buffer containing input data
  * @param out Buffer of photometrically corrected data
  */
-void phoCalWithBackplane ( std::vector<Isis::Buffer *> &in, std::vector<Isis::Buffer *> &out ) {
-
-  Buffer &image = *in[0];
-  Buffer &phase = *in[1];
-  Buffer &emission = *in[2];
-  Buffer &incidence = *in[3];
-  Buffer &calibrated = *out[0];
-
-  for (int i = 0; i < image.size(); i++) {
-    //  Don't correct special pixels
-    if (IsSpecial(image[i])) {
-      calibrated[i] = image[i];
-    }
-    else {
-    // Get correction and test for validity
-      double ph = g_pho->photometry(incidence[i], emission[i], phase[i], image.Band(i));
-      calibrated[i] = (IsSpecial(ph) ? Null : image[i] * ph);
-    }
-  }
-  return;
-}
+ void phoCal(Buffer &in, Buffer &out){
+   // Iterate through pixels
+   for(int i = 0; i < in.size(); i++){
+     // Ignore special pixels
+     if(IsSpecial(in[i])){
+       out[i] = in[i];
+     }
+     else{
+       // Get correction and test for validity
+       double ph = g_pho->compute(in.Line(i), in.Sample(i), in.Band(i), g_useDEM);
+       out[i] = ( IsSpecial(ph) ? Null : (in[i] * ph) );
+     }
+   }
+   return;
+ }//end phoCal
+
+ /**
+  * @brief Apply LROC Empirical photometric correction with backplane
+  *
+  * Short function dispatched for each line to apply the LROC Empirical photometrc
+  * correction function.
+  *
+  * @author 2016-09-19 Victor Silva
+  *
+  * @internal
+  *   @history 2016-09-19 Victor silva - Adapted from lrowacpho written by Kris Becker
+  *
+  * @param in Buffer containing input data
+  * @param out Buffer of photometrically corrected data
+  */
+ void phoCalWithBackplane ( std::vector<Isis::Buffer *> &in, std::vector<Isis::Buffer *> &out ) {
+
+   Buffer &image = *in[0];
+   Buffer &phase = *in[1];
+   Buffer &emission = *in[2];
+   Buffer &incidence = *in[3];
+   Buffer &calibrated = *out[0];
+
+   for (int i = 0; i < image.size(); i++) {
+     //  Don't correct special pixels
+     if (IsSpecial(image[i])) {
+       calibrated[i] = image[i];
+     }
+     else {
+     // Get correction and test for validity
+       double ph = g_pho->photometry(incidence[i], emission[i], phase[i], image.Band(i));
+       calibrated[i] = (IsSpecial(ph) ? Null : image[i] * ph);
+     }
+   }
+   return;
+ }
+
+
+ /**
+  * This method returns an QString containing the path of an
+  * LRO calibration directory
+  *
+  * @param calibrationType
+  * @param calibrationDirectory Path of the calibration directory
+  *
+  * @internal
+  *   @history 2021-03-12 Victor Silva - Added option for base calibration directory
+  */
+ void GetCalibrationDirectory(QString calibrationType, QString &calibrationDirectory) {
+   PvlGroup &dataDir = Preference::Preferences().findGroup("DataDirectory");
+   QString missionDir = (QString) dataDir["LRO"];
+   if(calibrationType != "")
+     calibrationType += "/";
+
+   calibrationDirectory = missionDir + "/calibration/" + calibrationType;
+ }
```

Comment:
If you swap to how hical handles this you don't need to worry about this. If you want to stick with this setup, then you can just use FileName::expanded to expand out `$lro/calibration` to the LRO calibration directory.

Reply:
Swapped it out. Thanks!